pongo2_template_test.go

```diff
@@ -344,10 +344,28 @@ func TestTemplate_Functions(t *testing.T) {
 			errorMessage: "[Error (where: execution) in <string> | Line 1 Col 4 near 'testFunc'] function input argument 0 of 'testFunc' must be of type int or *pongo2.Value (not <nil>)",
 			wantErr:      true,
 		},
+		{
+			name:     "KeywordArguments",
+			template: `{{ testFunc(arg1=1, isEnabled= 2 > 3, "no named arg", cached=true, 10) }}`,
```

Comment:
You are mixing named and unnamed arguments. Can we just copy Python conventions here.
1) When using keyword arguments, it is not necessary to follow the order of formal arguments in function definition.
2) In Python, the positional arguments must be before the keyword arguments while using mixed calling. Same here, however, we will rellok at this below.
3) As the Positional argument cannot appear after keyword arguments, Python raises the following error message −
    division(num=5, 10)
                      ^
    SyntaxError: non-keyword arg after keyword arg
4) "testFunc" should be implemented with signature `func(kwargs map[string]*pongo2.Value, args ...*pongo2.Value) string {`
     Function signature needs all named arguments passed as a dictionary. It also needs to be before var-args. In general, we can mandate a function signature where the order would be (non_var_arg1 *pongo2.Value, non_var_arg2 *pongo2.Value, keyword_args map[string](*pongo2.Value), ... all_var_args_array [](*pongo2.Value))

Reply:
Regarding `"testFunc" should be implemented with signature func(kwargs map[string]*pongo2.Value, args ...*pongo2.Value) string` python expects **kwargs, not a map. Python is able support **kwargs because it allows keyword arguments itself while go does not support.

If we pass map, then we dont need to add any new code at all. But problem is that then we will have to pass nil as first arg for all current instances of DeRef(nil, "models/x") which we do not want.

Reply:
We won't have to change the current instances to DeRef(nil, "models/x"). Pongo would be responsible for creating the map and populating it, if any kwargs are there. so call site world be same DeRef("models/x"). The DeRef function signature in go lang would change.